visma/matrix/structure.py

```diff
@@ -103,22 +131,37 @@ def determinant(self):
         pass
 
 
-class IdenMat(SquareMat):
-    """Class for identity matrix
+class DiagMat(SquareMat):
+    """Class for Diagonal matrix
 
-    Identity matrix is a square matrix with all elements as 0 except for the diagonal elements which are 1.
+    Diagonal matrix is a square matrix with all elements as 0 except for the diagonal elements.
 
     Extends:
         SquareMat
     """
 
-    def __init__(self, dim):
+    def __init__(self, dim, diagElem):
```

Comment:
Is `diagElem` a list of int here?
Update the `self.dim = dim` before line 145.

If possible try to pass `diagElem`  as a list of list of tokens and add a docstring/comment in  `__init__` defining the argument types.